c++/src/Compression.cc

```diff
@@ -1203,8 +1237,8 @@ namespace orc {
       case CompressionKind_ZLIB: {
         int level =
             (strategy == CompressionStrategy_SPEED) ? Z_BEST_SPEED + 1 : Z_DEFAULT_COMPRESSION;
-        return std::make_unique<ZlibCompressionStream>(outStream, level, bufferCapacity,
-                                                       compressionBlockSize, pool, metrics);
+        return std::make_unique<ZlibCompressionStream>(
+            outStream, level, bufferCapacity, compressionBlockSize, memoryBlockSize, pool, metrics);
```

Comment:
If `memoryBlockSize` is only used for streaming compressor, we should rename it to be clear. However, I do think we need to support block compressor as well as ZSTD (block-based) is the default one. This can be a separate PR though.

Reply:
Good advice. I will do this in my next work if necessary